src/plugins/azgenerator/AzExtensionIncrementalGenerator.ts

```diff
@@ -14,40 +17,49 @@ import { GenerateAzureCliParams } from "./TemplateAzureCliParams";
 import { GenerateAzureCliTestPrepare } from "./TemplateAzureCliTestPrepare";
 import { GenerateAzureCliTestScenario, NeedPreparer } from "./TemplateAzureCliTestScenario";
 import { GenerateAzureCliValidators } from "./TemplateAzureCliValidators";
+import { CliTopCustom } from "./templates/CliTopCustom";
+import { CliTopHelp } from "./templates/CliTopHelp";
+import { CliTopInit } from "./templates/CliTopInit";
 
 export class AzExtensionIncrementalGenerator extends AzGeneratorBase {
-
-    constructor(model: CodeModelAz, isDebugMode: boolean) {
-        super(model, isDebugMode);
-        this.path = "azext_" + this.model.Extension_NameUnderscored + "/";
+    constructor(host: Host, model: CodeModelAz, isDebugMode: boolean) {
+        super(host, model, isDebugMode);
```

Comment:
why do you need host ?

Reply:
I think we can complete the read/write in Generator. Previously, we only generate the code. but for incremental codegen, we need generate base on existing code. If we put the read/write code outside. It's dup logic for all generator. So I put the host here, which will pass to each file genreator in generateAll()

---
In reply to: [472063087](https://github.com/Azure/autorest.az/pull/519#discussion_r472063087) [](ancestors = 472063087)